Tools/clinic/clinic.py

```diff
@@ -5580,12 +5577,6 @@ def do_post_block_processing_cleanup(self) -> None:
             if no_parameter_after_star:
                 fail("Function " + self.function.name + " specifies '*' without any parameters afterwards.")
 
-        # remove trailing whitespace from all parameter docstrings
-        for name, value in self.function.parameters.items():
-            if not value:
-                continue
-            value.docstring = value.docstring.rstrip()
```

Comment:
Do we have a test that ensures the parameter docstrings have no trailing whitespace? Seems like no tests fail if I apply just this diff to `main` -- was it just always redundant?

```diff
--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -5580,12 +5580,6 @@ def do_post_block_processing_cleanup(self) -> None:
             if no_parameter_after_star:
                 fail("Function " + self.function.name + " specifies '*' without any parameters afterwards.")

-        # remove trailing whitespace from all parameter docstrings
-        for name, value in self.function.parameters.items():
-            if not value:
-                continue
-            value.docstring = value.docstring.rstrip()
-
         self.function.docstring = self.format_docstring()
```

Reply:
> was it just always redundant?

No, it was just untested, not redundant: a test _does_ fail if I apply _this_ diff to `main`:

```diff
--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -5584,7 +5584,7 @@ def do_post_block_processing_cleanup(self) -> None:
         for name, value in self.function.parameters.items():
             if not value:
                 continue
-            value.docstring = value.docstring.rstrip()
+            assert value.docstring == value.docstring.rstrip()

         self.function.docstring = self.format_docstring()
```

(`ClinicParserTest.test_function_docstring`)

Reply:
I'll add a test :)